undertow/src/main/java/org/wildfly/extension/undertow/Server.java

```diff
@@ -169,9 +172,24 @@ public String getDefaultHost() {
     }
 
     public String getRoute() {
-        UndertowService service = this.undertowService.getValue();
-        String defaultServerRoute = service.getInstanceId();
-        return this.name.equals(service.getDefaultServer()) ? defaultServerRoute : String.join("-", defaultServerRoute, this.name);
+        final UndertowService service = this.undertowService.getValue();
+        final String defaultServerRoute = service.getInstanceId();
+        final String route = this.name.equals(service.getDefaultServer()) ? defaultServerRoute : String.join("-", defaultServerRoute, this.name);
+        if (service.isObfuscateInstanceId()) {
+            try {
+                final MessageDigest md = MessageDigest.getInstance("MD5");
+                // salt
+                md.update(this.name.getBytes());
```

Comment:
Given that we don't know the default character encoding of a given host, it's probably best to keep this consistent across all hosts, e.g.
```java
md.update(this.name.getBytes(StandardCharsets.UTF-8));
```

Reply:
noted